assets/shared/helpers/player/use-editor-player.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import Player from './index';
+
+const API_SCRIPT_ID = 'player-script';
+const YOUTUBE_API_SRC = 'https://www.youtube.com/iframe_api';
+const VIMEO_API_SRC = 'https://player.vimeo.com/api/player.js';
+
+/**
+ * Hook to get the dependencies used as trigger to (re)set the player instance.
+ *
+ * @param {Object} videoBlock Video block object.
+ *
+ * @return {Object} Object containing embed fetching and block selected state.
+ */
+const useTriggerDependencies = ( videoBlock ) => {
+	// Check embed fetching.
+	const { fetching } = useSelect(
+		( select ) => ( {
+			fetching: select( coreStore ).isRequestingEmbedPreview(
+				videoBlock?.attributes?.url
+			),
+		} ),
+		[ videoBlock?.attributes?.url ]
+	);
+
+	// Check if block is selected. We need to get the player reference again when it's video block
+	// because it re-creates the video element when it's (un)selected.
+	const { isBlockSelected } = useSelect(
+		( select ) => ( {
+			isBlockSelected: select( blockEditorStore ).isBlockSelected(
+				videoBlock.clientId
+			),
+		} ),
+		[ videoBlock.clientId ]
+	);
+
+	return { fetching, isBlockSelected };
+};
+
+/**
+ * A wrapper to useEffect with a timeout of 1 milisecond, in order to delay the sync execution.
```

Comment:
If we only need to make this async, could we just use `setTimeout` without any delay? That should just defer it to later in the event loop, right?

Reply:
Exactly! Updated to `0` https://github.com/Automattic/sensei/pull/5377/commits/2db6bb09d11e48dfa486ea159df104be88ca6a64

Reply:
Nice! I think omitting the argument would also work (default value is `0`), but an explicit `0` is also good 👍

Reply:
Ah! That's true! Updated here: https://github.com/Automattic/sensei/pull/5377/commits/57a9c52fcf54136cef83ae23449073d042fa3769

Thank you for the suggestion! I used `1` my whole life! Time to change! haha